### DISCUSSION TITLE
missing lead hours in output stat file name

### DISCUSSION THREAD

**Discussion by JiliDong-NOAA** [dtcenter/METplus#1321] [File I/O] [answered]
I am using metplus to generate ensemble stats and found out the output stats file name has no lead hours, e.g.
ensemble_stat_rrfs_ADPSFC_NDAS_20210504_180000V.stat

The deterministic stats do have the lead hours in the file name, e,g,:
point_stat_RRFS_CLOUD_HWT_NDAS_ADPSFC_360000L_20210504_120000V.stat

I wonder if there are parameters/options I should change to have it back for ensemble stats.

Thanks for help.

Jili

**Comment by j-opatz** [comment, marked as answer]
Hi Jili,

And thank you for your question. Ensemble-Stat does not list the lead hours by default, instead using the valid time. In the [MET user's guide](https://met.readthedocs.io/en/develop/Users_Guide/ensemble-stat.html#ensemble-stat-output), there's the following explanation for this default choice:

> The output STAT file is named using the following naming convention:
> 
> ensemble_stat_PREFIX_YYYYMMDD_HHMMSSV.stat where PREFIX indicates the user-defined output prefix and YYYYMMDD_HHMMSSV indicates the forecast valid time. Note that the forecast lead time is not included in the output file names since it would not be well-defined for time-lagged ensembles. 

If it's important that you have the lead time in the file name, you can utilize the "output_prefix" option to customize the file names as needed. Information on that option is [here](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html?highlight=%22output_prefix%22#output-prefix).

I hope this helps answer your question; if it has, please mark this as the answer and I'll go ahead and lock the discussion thread. If you need more clarification on this question, please reply as needed.

**Reply by JiliDong-NOAA** [reply]
Thanks for the explanation. That makes sense. I am able to get lead hours by including them in output_prefix. Will mark this as answer and thanks again for help!
